### PR TITLE
Added wait_for_tcp for ldap

### PR DESCRIPTION
**Pull request by rhpmali**
@xingxingxia @barleyer please review. Added wait_for_tcp for ldap

Job link: Runner-v3-smoke/2497/console

### CODE REVIEW

File: features/step_definitions/helper_services.rb [hunk 1]
```diff
@@ -179,14 +179,23 @@
     # Init the test data in ldap server.
     @result = pod.exec("ldapadd", "-x", "-h", "127.0.0.1", "-p", "389", "-D", "cn=Manager,dc=example,dc=com", "-w", "admin", stdin: File.read(cb.test_file), as: user)
     step %Q/the step should succeed/
-
+
```
Comment (xingxingxia) [inline features/step_definitions/helper_services.rb:182]: Unneeded change here.

File: features/step_definitions/helper_services.rb [hunk 2]
```diff
@@ -179,14 +179,23 @@
...
     step %Q/the step should succeed/
+
+    @result = BushSlicer::Common::Net.wait_for_tcp(host: "localhost", port: cb.ldap_port, timeout: 60)
+    stats = @result[:props][:stats]
+    logger.info "after #{stats[:seconds]} seconds and #{stats[:iterations]} " <<
+      "iterations, localhost is: " <<
+      "#{@result[:success] ? "accessible" : @result[:error].inspect}"
+
+    raise "port forwarding is not yet ready" unless @result[:success]
+    step %Q/the step should succeed/
```
Comment (xingxingxia) [inline features/step_definitions/helper_services.rb:198]: Here dup check of step %Q/the step should succeed/

**Comment by xingxingxia** [comment]
LGTM except the minor comments.

**Comment by rhpmali** [comment]
Modified as per suggestion, please review

**Comment by akostadinov** [comment]
Very nice!

**Pull request merged**